tutorials/R/spatial-data/spatialData.Rmd

```diff
@@ -22,331 +23,514 @@ connecting remote sensing with ground-based measurements, the methods in
+
+The Field Site location data is also available in as a Shapefile and KMZ on the 
+Spatial Data and Maps page. We use the file from the site list to demonstrate 
+alternative ways to work with spatial data. 
+
+## Map NEON domains 
+
+Using the domain shapefile and the field sites list, let's make a map of NEON 
+site locations.
 
 We'll read in the spatial data using the `rgdal` and `sp` packages 
 and plot it using the `ggplot2` package. First, read in the domain 
-shapefile:
+shapefile. 
 
-``` {r domains, results="hide"}
+Be sure to move the downloaded data files into the  working directory you set 
+earlier!
 
-# modify "~/data" to the filepath where you downloaded the shapefile
-neon.domains <- readOGR("~/data/NEONDomains_0", layer="NEON_Domains")
+```{r load-domains, results="hide"}
 
-# the next two commands convert the shapefile to a format ggplot 
-# can use
-neon.domains <- SpatialPolygonsDataFrame(gSimplify(neon.domains, tol=0.1, 
-                                                 topologyPreserve=TRUE), 
-                               data=neon.domains@data)
-map <- fortify(neon.domains, region="DomainName")
+# upload data
+neonDomains <- readOGR("NEONDomains_0" , layer="NEON_Domains")
+```
+
+The data come is as a Large SpatialPolygonsDataFrame, which unfortunately, ggplot
+can't use. Therefore, we'll need to make a few changes to the data structure to 
+convert it to a DataFrame that ggplot can use. 
+
+```{r create-df, warning=FALSE}
+
+# First, add a new column termed "id" composed of the row names of the data
+neonDomains@data$id <- rownames(neonDomains@data)
+
+# Now, use tidy() to convert to a dataframe
+# if you previously used fortify(), this does the same thing. 
+neonDomains_points<- tidy(neonDomains, region="id")
 
+# Finally, merge the new data with the data from our spatial object
+neonDomainsDF <- merge(neonDomains_points, neonDomains@data, by = "id")
 ```
+Now that the data are in a dataframe, lets check out what data are available 
+for us to plot
 
-Let's plot the domains without the sites first:
+```{r explore-domains}
+# view data structure for each variable
+str(neonDomainsDF)
 
-``` {r plot-domains, message=F, warning=F, fig.width=8, fig.height=6}
+```
+
+We do have **long**itude and **lat**itude data, as well as names of the domains. 
+With this info and now that we have the data in a data frame, we can plot it in 
+ggplot. Let's first plot the domains without the sites. Since we a plotting a 
+map, this means our longitude is plotted on the x axis and latitude is on our 
+y axis. 
 
-gg <- ggplot() + theme_map()
-gg <- gg + geom_map(data=map, map=map,
-                    aes(x=long, y=lat, map_id=id, group=group),
-                    fill="white", color="black", size=0.3)
-gg
+```{r plot-domains, fig.width=8, fig.height=6}
+# plot domains
+domainMap <- ggplot(neonDomainsDF) + 
+        geom_map(map = neonDomainsDF,
+                aes(x = long, y = lat, map_id = id),
```

Comment:
This is an area I'm not 100% happy with.  As written get warning "Warning message: Ignoring unknown aesthetics: x, y ", in addition geom_map says nothing about including x & y.  But if not included, plot doesn't work.   Tried to find a solution for too long so just leaving as is for now.

Reply:
Yes, I had the same problem when I set it up in the first place, it's totally maddening! I never found a solution either.